plugins/PreviewStage/PreviewMenu.qml

```diff
@@ -15,80 +15,127 @@ Item
-            // scene).
-            id: viewModeButton
+            anchors.centerIn: parent
+            //spacing: UM.Theme.getSize("default_margin").width
+            height: parent.height
 
-            style: UM.Theme.styles.combobox
+            Cura.ExpandableComponent
+            {
+                id: viewSelector
+                iconSource: expanded ? UM.Theme.getIcon("arrow_bottom") : UM.Theme.getIcon("arrow_left")
+                height: parent.height
 
-            model: UM.ViewModel { }
-            textRole: "name"
+                property var viewModel: UM.ViewModel { }
 
-            // update the model's active index
-            function updateItemActiveFlags()
-            {
-                currentIndex = getActiveIndex()
-                for (var i = 0; i < model.rowCount(); i++)
+                property var activeView:
                 {
-                    model.getItem(i).active = (i == currentIndex)
+                    for (var i = 0; i < viewModel.rowCount(); i++)
+                    {
+                        if(viewModel.getItem(i).active)
+                        {
+                            return viewModel.getItem(i)
+                        }
+                    }
+                    // Nothing was active, so just return the first one (the list is sorted by priority, so the most
+                    // important one sshould be returned)
+                    return viewModel.getItem(0)
                 }
-            }
 
-            // get the index of the active model item on start
-            function getActiveIndex()
-            {
-                for (var i = 0; i < model.rowCount(); i++)
+                // Ensure that the controller is synced with whatever happend here.
+                onActiveViewChanged: UM.Controller.setActiveView(activeView.id)
+
+                headerItem: Label
                 {
-                    if (model.getItem(i).active)
-                    {
-                        return i;
-                    }
+                    text: viewSelector.activeView.name
+                    verticalAlignment: Text.AlignVCenter
+                    height: parent.height
+                    elide: Text.ElideRight
+                    font: UM.Theme.getFont("default")
                 }
-                return 0
-            }
 
-            onCurrentIndexChanged:
-            {
-                if (model.getItem(currentIndex).id != undefined)
+                popupItem: Column
                 {
-                    UM.Controller.setActiveView(model.getItem(currentIndex).id)
+                    id: column
+                    width: viewSelector.width - 2 * UM.Theme.getSize("default_margin").width
+
+                    // For some reason the height of the column gets set to 0 if this is not set...
+                    Component.onCompleted: height = implicitHeight
+
+                    Repeater
+                    {
+                        id: networkedPrinters
```

Comment:
I'm sure this is not a good ID for this component. It's the list of views, right?
```suggestion
                        id: viewsList
```